src/jobs/test.yml

```diff
@@ -48,27 +60,83 @@ steps:
+            working_directory: <<parameters.app-dir>>
+            command: npm list jest-junit || (echo "Add the package jest-junit to the dev dependencies of your project with `npm install --save-dev jest-junit`" && exit 1)
+        - run:
+            name: Run NPM Tests
+            working_directory: <<parameters.app-dir>>
+            command: npm run <<parameters.run-command>> -- --reporters=default --reporters=jest-junit
+  - when: # Run tests for YARN, without test results
       condition:
-        equal: [yarn, << parameters.pkg-manager >>]
+        and:
+          - equal: [yarn, << parameters.pkg-manager >>]
+          - equal: [none, << parameters.test-results-for >>]
       steps:
         - run:
             name: Run YARN Tests
             working_directory: <<parameters.app-dir>>
             command: yarn run <<parameters.run-command>>
-  - when: # Run tests for YARN 2.x
+  - when: # Run tests for YARN, with jest test results
       condition:
-        equal: [yarn-berry, << parameters.pkg-manager >>]
+        and:
+          - equal: [yarn, << parameters.pkg-manager >>]
+          - equal: [jest, << parameters.test-results-for >>]
+      steps:
+        - run:
+            name: Check for test reporter
+            working_directory: <<parameters.app-dir>>
+            command: yarn list --pattern jest-junit | grep " jest-junit@" || (echo "Add the package jest-junit to your projects dev dependencies with `yarn add --dev jest-junit`" && exit 1)
+        - run:
+            name: Run YARN Tests
+            working_directory: <<parameters.app-dir>>
+            command: yarn run <<parameters.run-command>> --reporters=default --reporters=jest-junit
+  - when: # Run tests for YARN 2.x, without test results
+      condition:
+        and:
+          - equal: [yarn-berry, << parameters.pkg-manager >>]
+          - equal: [none, << parameters.test-results-for >>]
       steps:
         - run:
             name: Run YARN 2.x Tests
             working_directory: <<parameters.app-dir>>
             command: yarn run <<parameters.run-command>>
+  - when: # Run tests for YARN 2.x, with jest test results
+      condition:
+        and:
+          - equal: [yarn-berry, << parameters.pkg-manager >>]
+          - equal: [jest, << parameters.test-results-for >>]
+      steps:
+        - run:
+            name: Run YARN 2.x Tests
+            working_directory: <<parameters.app-dir>>
+            command: yarn run <<parameters.run-command>> --reporters=default --reporters=jest-junit
+  - when: # upload test results if a path was given
+      condition: << parameters.test-results-path >>
+      steps:
+        store_test_results:
+          path: << parameters.test-results-path >>
+  - when: # upload test results if it was automatically configured
+      condition:
+        not:
+          equal: [none, << parameters.test-results-for >>]
+      steps:
+        store_test_results:
+          path: <<parameters.app-dir>>/junit.xml
```

Comment:
Might users still possibly have a different file name for their junit output? will the app dir always be where the results are stored?

Reply:
Let me make this more specific for when `jest` is selected and test-results-path is not specified. This was intended to cover jest-junit default file location and name.

Reply:
That makes sense! Thanks 👍